src/puppetlabs/puppetdb/scf/storage.clj

```diff
@@ -793,6 +824,9 @@
     received-timestamp :- pls/Timestamp]
    (time! (:replace-catalog performance-metrics)
           (jdbc/with-db-transaction []
+            ;; TODO: figure out if we want/need to include catalog inputs in the similarity hash for catalogs
+            ;; this could cause issues similar to those with package inventory wrt sync
+            ;; it might be better to have catalog inputs sync as a diff command with each catalog entity in sync
```

Comment:
Yes we probably should not hash the catalog inputs into the catalog hash because that'd mean we'd need to query/reconstruct the inputs on every catalog submission that didn't include them.

This does present sync difficulties that may result in needing a new `"replace catalog-inputs"` command type